docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,197 @@
+
+*Accessible from a central UI/API*
+
+Platform Admin or Storage Admin can visualize logs from all containers from all
+namespaces as well as journal logs, including (kubelet, containerd, salt-minion
+, kernel, initrd, services, etc ...) in Grafana. One can correlate logs and
+metrics or alerts in one single Grafana Dashboard.
+Browsing logs can be achieved through a documented API in order to expose logs
+in MetalK8s UIs or other workloads UIs if needed.
+
+*Persistence, Retention*
+
+Logs should be stored on a persistent storage. Platform Administrator can
+configure a max retention period. Some automatic purging mechanism is triggered
+when logs are older than the retention period or when the persistent storage is
+about to reach its capacity limit.
+A typical and default retention period is of 2 weeks.
+
+*Horizontally scalable (capacity and performance)*
+
+The Platform Administrator can add nodes or disks in order to increase storage
+capacity for logs (to keep more history or to store more workloads logs) or to
+increase performances for ingesting or querying the logs. It is also possible
+to fine tune the deployment configuration in order to have more replicas and
+ideally the Log centralisation service supports horizontal POD autoscaling.
+
+*Highly Available*
+
+Log collection, storage and browsing services can be replicated in order to
+ensure the same level of High Availability Prometheus has i.e. can lose all
+replicas but one.
+
+*Log Querying*
+
+Get all logs for a given period, node(s), pod regex, limited list of predefined
+labels and free keywords text.
+Typical Zenko use case: collecting all logs across several components, related
+to a S3 uniq request.
+
+*Log statistics (nice to have)*
+
+The Log centralisation service also offer the ability to consume statistics
+about the logs like the number of occurrences of one type of log during a
+certain period of time.
+
+*Monitorable/Observable service (health, performances and alerts)*
+
+The Platform Administrator can monitor capacity usage, IOPS, latency and
+bandwidth of the Log centralisation service. He can also monitor the health of
+the service (i.e. if some active alerts exist). He can visualize active and
+past alerts impacting the service.
+
+*Performances (TBD)*
+
+Typical workloads can generate around 1000 logs per second per node.
+
+User Stories
+------------
+
+- As a Platform Administrator, I want to browse all MetalK8s containers logs
+  (from all servers) from a unique endpoint, in order to ease distributed
+  k8s and metalk8s services error investigation.
+
+- As a Platform Administrator, I want to browse non container (kubelet,
+  containerd, salt-minion, cron) logs (from all servers) from a
+  unique endpoint, in order to ease System error investigation.
+
+- As a Storage Administrator, I want to browse all Solution instance containers
+  logs (from all servers) from a unique endpoint, in order to ease Solution
+  instance Error investigation.
```

Comment:
```suggestion
  instance error investigation.
```

Would need to express the filtering/search capabilities in the User Story as well, IMHO (make it clear that, as a Storage Admin, I can easily drill down into the affected workloads that I want logs from).